content/tutorial/tutorial.md

```diff
@@ -1170,41 +1177,41 @@ We will also replace reading `this.state.history` with `this.state.history.slice
 
 ```javascript{3}
   render() {
     const history = this.state.history;
     const current = history[this.state.stepNumber];
     const winner = calculateWinner(current.squares);
 
-    // the rest has not changed
+    // और इसके आगे कोई चाल बाकि नहीं है
 ```
 
-If we click on any step in the game's history, the tic-tac-toe board should immediately update to show what the board looked like after that step occurred.
+यदि हम गेम के history के किसी भी चरण पर क्लिक करते हैं, तो टिक-टैक-टो Board को तुरंत यह दिखाने के लिए अपडेट करना चाहिए कि Board उस चरण के बाद कैसा दिखता था।
 
-**[View the full code at this point](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**
+**[अभी तक का पूरा कोड देखें](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**
 
-### Wrapping Up {#wrapping-up}
+### अंतिम स्टेप्स {#wrapping-up}
 
-Congratulations! You've created a tic-tac-toe game that:
+बधाई हो! आपने एक टिक-टैक-टो गेम बनाया है जो:
 
-* Lets you play tic-tac-toe,
-* Indicates when a player has won the game,
-* Stores a game's history as a game progresses,
-* Allows players to review a game's history and see previous versions of a game's board.
+* अब टिक-टैक-टो खेलते हैं,
+* संकेत करता है कि जब किसी खिलाड़ी ने गेम जीता है,
+* एक खेल के रूप में एक खेल के इतिहास को स्टोर करता है,
+* खिलाड़ियों को गेम के इतिहास की समीक्षा करने और गेम के बोर्ड के पिछले संस्करणों को देखने की अनुमति देता है।
 
-Nice work! We hope you now feel like you have a decent grasp on how React works.
+अच्छा काम! हमें उम्मीद है कि अब आप महसूस करेंगे कि आपके पास React कैसे काम करता है, इस पर एक सभ्य समझ है।
 
-Check out the final result here: **[Final Result](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**.
+अंतिम परिणाम यहां देखें: **[अंतिम परिणाम](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**.
 
-If you have extra time or want to practice your new React skills, here are some ideas for improvements that you could make to the tic-tac-toe game which are listed in order of increasing difficulty:
+यदि आपके पास अतिरिक्त समय है या आप अपने नए React कौशल का अभ्यास करना चाहते हैं, तो यहां सुधार के लिए कुछ विचार दिए गए हैं जो आप टिक-टैक-टो खेल में कर सकते हैं जो कि बढ़ती कठिनाई के क्रम में सूचीबद्ध हैं:
 
-1. Display the location for each move in the format (col, row) in the move history list.
-2. Bold the currently selected item in the move list.
-3. Rewrite Board to use two loops to make the squares instead of hardcoding them.
-4. Add a toggle button that lets you sort the moves in either ascending or descending order.
-5. When someone wins, highlight the three squares that caused the win.
-6. When no one wins, display a message about the result being a draw.
+1. चाल इतिहास सूची में प्रारूप (col, row) में प्रत्येक चाल के लिए स्थान प्रदर्शित करें।
+2. चाल सूची में वर्तमान में चयनित आइटम को बोल्ड करें।
+3. स्क्वायरस को हार्डकोड करने के बजाय दो छोरों का उपयोग करने के लिए रीराइट बोर्ड का उपयोग करें।
```

Comment:
छोरों instead of loops doesn't make much sense?

Reply:
> छोरों instead of loops doesn't make much sense?

`लूप्स`